src/software/firmware/includes/debug.h

```diff
@@ -91,44 +93,99 @@
 #endif
 
 /**
- * Send metrics to serial
+ * Send data metrics to serial ( almost evrey 1/100s) 
+ * Take notice that sending 30 bytes of data can take up to 3ms (1/3 of the 10ms loop) at a 115200 baudrate.
+ * Please consider reducing CPU charge using a DMA on the UART TX.
  *
  * @param cycle            Number of passed cycles
  * @param centieme         Duration from the begining of the cycle in hundredth of second
  * @param periode
  * @param phase            Current phase
  * @param subPhase         Current subphase
  * @param pression         Current pressure
- * @param consigneBlower   Requested aperture for the blower Pressure Valve
  * @param positionBlower   Current aperture of the blower Pressure Valve
- * @param consignePatient  Requested aperture for the patient Pressure Valve
  * @param positionPatient  Current aperture of the patient Pressure Valve
+ * @param blowerSpeed      Current speed of the blower
+ * @param baterryLevel     Current battery level if any
  */
 #if DEBUG == 1
-#define DBG_PHASE_PRESSION(cycle, centieme, periode, phase, subPhase, pression, consigneBlower,    \
-                           positionBlower, consignePatient, positionPatient)                       \
+#define DBG_TELEMERTY_DATA(cycle, centieme, periode, phase, subPhase, pression,    \
+                           positionBlower, positionPatient, blowerSpeed, batteryLevel)                       \
     if ((centieme % periode) == 0u) {                                                              \
-        Serial.print((cycle * 300u) + centieme);                                                   \
-        Serial.print(";");                                                                         \
+        Serial.print("D"); \
```

Comment:
Could you try using `Serial.printf("D\tUID [HEX] : %x %x %x\t...",*(uint32_t*)(UID_BASE),...);`
This would avoid calling multiple `Serial.print`. Only one call could performed with the right formatting

Reply:
Let us (@ABOSTM  and me) know if this improve the MCU load.

Reply:
1st point, is it really mandatory to send those data every cycle, every 10ms ? This would be a huge amont of data. Maybe it would be possible to reduce data (send data only in case of issue; or 1 time out of 2 , ...), it is then maybe possible to increase transmission frequency. 
And then yes DMA is a possibility but not with Arduino API, and it is much more comple  see my answer about I2C: https://makersforlife.slack.com/archives/C010K0FHYHL/p1586850747470300?thread_ts=1586813725.463400&cid=C010K0FHYHL